x-pack/plugins/observability_solution/slo/server/services/get_slo_groupings.ts

```diff
@@ -0,0 +1,157 @@
+    sloId: string,
+    params: GetSLOGroupingsParams
+  ): Promise<GetSLOGroupingsResponse> {
+    const { slo } = await this.definitionClient.execute(sloId, this.spaceId, params.remoteName);
+
+    const groupingKeys = [slo.groupBy].flat();
+    if (groupingKeys.includes(ALL_VALUE) || params.instanceId === ALL_VALUE) {
+      throw new Error('Ungrouped SLO cannot be queried for available groupings');
+    }
+
+    if (!groupingKeys.includes(params.groupingKey)) {
+      throw new Error("Provided groupingKey doesn't match the SLO's groupBy field");
+    }
+
+    const groupingValues = params.instanceId.split(',') ?? [];
+    if (groupingKeys.length !== groupingValues.length) {
+      throw new Error('Provided instanceId does not match the number of grouping keys');
+    }
+
+    const response = await this.esClient.search<
+      unknown,
+      {
+        groupingValues: {
+          buckets: Array<{ key: { value: string } }>;
+          after_key: { value: string };
+        };
+      }
+    >({
+      index: params.remoteName
+        ? `${params.remoteName}:${SLO_SUMMARY_DESTINATION_INDEX_NAME}`
+        : SLO_SUMMARY_DESTINATION_INDEX_NAME,
+      ...generateQuery(slo, params, this.sloSettings),
+    });
+
+    return {
+      groupingKey: params.groupingKey,
+      values: response.aggregations?.groupingValues.buckets.map((bucket) => bucket.key.value) ?? [],
+      afterKey:
+        response.aggregations?.groupingValues.buckets.length === Number(params.size ?? DEFAULT_SIZE)
+          ? response.aggregations?.groupingValues.after_key.value
+          : undefined,
+    };
+  }
+}
+
+function generateQuery(slo: SLODefinition, params: GetSLOGroupingsParams, settings: SLOSettings) {
+  const groupingKeys = [slo.groupBy].flat();
+  const groupingValues = params.instanceId.split(',') ?? [];
+
+  const groupingKeyValuePairs = groupingKeys.map((groupingKey, index) => [
+    groupingKey,
+    groupingValues[index],
+  ]);
+
+  const aggs = generateAggs(params);
+
+  const query = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            term: {
+              'slo.id': slo.id,
+            },
+          },
+          {
+            term: {
+              'slo.revision': slo.revision,
+            },
+          },
+          // exclude stale summary documents if specified
+          ...(!!params.excludeStale
+            ? [
+                {
+                  range: {
+                    summaryUpdatedAt: {
+                      gte: `now-${settings.staleThresholdInHours}h`,
+                    },
+                  },
+                },
+              ]
+            : []),
+          // Set other groupings as term filters
+          ...groupingKeyValuePairs
+            .filter(([groupingKey]) => groupingKey !== params.groupingKey)
+            .map(([groupingKey, groupingValue]) => ({
+              term: {
+                [`slo.groupings.${groupingKey}`]: groupingValue,
+              },
+            })),
```

Comment:
💬 We filter the summary documents to match the other grouping key/value pairs. So the values returned for the specified groupingKey form a valid instanceId when set together with the other pairs